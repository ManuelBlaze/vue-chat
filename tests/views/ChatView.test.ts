import { mount } from '@vue/test-utils';
import ChatMessages from '@/components/chat/ChatMessages.vue';
import ChatView from '@/views/ChatView.vue';
import MessageBox from '@/components/chat/MessageBox.vue';

const mockChatMessages = {
  template: '<div data-testid="mock-messages">Mock ChatMessages</div>',
};

describe('ChatView', () => {
  it('should properly renders chat messages and message box', () => {
    const wrapper = mount(ChatView);
    expect(wrapper.html).toMatchSnapshot();

    expect(wrapper.findComponent(ChatMessages).exists()).toBe(true);
    expect(wrapper.findComponent(MessageBox).exists()).toBe(true);
  });

  it('should emit onMessage when sending a message', async () => {
    const wrapper = mount(ChatView, {
      global: {
        stubs: {
          ChatMessages: mockChatMessages,
        },
      },
    });

    const messageBoxComponent = wrapper.findComponent(MessageBox);
    messageBoxComponent.vm.$emit('sendMessage', 'Wenas');

    await new Promise<void>((r) => setTimeout(r, 150));
    expect(wrapper.html).toMatchSnapshot();
  });
});
