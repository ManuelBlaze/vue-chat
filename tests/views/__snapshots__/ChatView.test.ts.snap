// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatView > should emit onMessage when sending a message 1`] = `[Function]`;

exports[`ChatView > should properly renders chat messages and message box 1`] = `[Function]`;
